### ISSUE TITLE
Any app built with Expo SDK 47 crashes on Android 7.1 or older

### ISSUE THREAD

**Issue by Gruak** [expo/expo#20031]
### Summary

Any app built with Expo SDK 47 crashes on splash screen when launched on Android 7.1 or older (API level <= 25) including Expo Go app.
I've tested both on physical devices and emulator with Android versions 7.1, 7 and 6 (crashes in each case), as well as on Android 8 and 13 (doesn't crash).
When I downgrade the expo package alone in package.json to expo@46 and build the app, it doesn't crash anymore.

This is the same case as reported in https://github.com/expo/expo/issues/20019 and https://github.com/expo/expo/issues/20024 but I've traced down the issue to be related with Android version and grouped them here.

### What platform(s) does this occur on?

Android

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 6.0 Manjaro Linux
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 18.12.1 - /run/user/1000/fnm_multishells/41703_1668416092989/bin/node
      Yarn: 1.22.19 - /run/user/1000/fnm_multishells/41703_1668416092989/bin/yarn
      npm: 9.1.1 - /run/user/1000/fnm_multishells/41703_1668416092989/bin/npm
    SDKs:
      Android SDK:
        API Levels: 23, 33
        Build Tools: 30.0.3, 33.0.0
        System Images: android-23 | Intel x86 Atom, android-33 | Google APIs Intel x86 Atom_64
    npmGlobalPackages:
      eas-cli: 2.7.0
      expo-cli: 6.0.8

### Minimal reproducible example

1. Launch Android emulator with Android 7.1 (API level 25)
2. Install and run the latest version of Expo Go on it

**Comment by RohovDmytro** [comment]
Same. It might be due to some module failing to init on older version of Android.

```
Fatal Exception: com.facebook.react.common.JavascriptException
Error: Exception in HostObject::get for prop 'NativeModulesProxy': java.lang.NoClassDefFoundError: Failed resolution of: Ljava/nio/file/Path;, js engine: hermes, stack: anonymous@1:1057937
````

**Comment by brentvatne** [comment]
install the latest version of the expo package for a fix: `expo@~47.0.5`

**Comment by otekdo** [comment]
@brentvatne We are hoping for a solution for this too, is pretty important.
For the first time, Many people are reporting that The Expo Go crashing at start
https://github.com/expo/expo/issues/20019

**Comment by brentvatne** [comment]
@otekdo - we can deploy this fix to expo go later, but as i said above, this is fixed in `@expo@~47.0.5`. it's a native change so you will need to rebuild/resubmit your app

**Comment by evelant** [comment]
@brentvatne this seems to have resurfaced in expo 47.0.9, my app is crashing on launch with exactly the same exception reported by the others above. So far it seems to only be happening on Android 6.0.1.
